### ISSUE TITLE
Remove wilcard redirect to mail.domain.tld when subdomain does not match

### ISSUE THREAD

**Issue by L0g4n** [Mailu/Mailu#1772]
I noticed that in the default setup when you use the integrated frontproxy that has configured tls certs for `mail.domain.tld` it also redirects to `domain.tld`. However, this produces scary warnings in the browser when your cert is only valid for `mail.domain.tld` (like in my case).

Is there already a config option for that available or do I have to modify the config manually to remove this behavior?

**Comment by Diman0** [comment]
This sounds like a misconfiguration. 
What version of Mailu are you using?
What settings did you use for `DOMAIN`, `HOSTNAMES` and `TLS_FLAVOR` in your mailu.env file?
As a reminder, these settings are described [here](https://mailu.io/1.7/configuration.html#common-configuration).

**Comment by L0g4n** [comment]
I'm using mailu version `1.7`.
DOMAIN: `domain.tld`
HOSTNAME:  `mail.domain.tld
TLS_FLAVOUR:  `cert`

**Comment by Diman0** [comment]
Those settings are fine. Where do you see that the admin page redirects you to domain.tld?
Note that the link for item website (in the web admin menu) points to the site configured for SITENAME in mailu.env

**Comment by L0g4n** [comment]
I think there was a misunderstanding. The redirection happens when you go to `domain.tld` -> It then wants to start a SSL Handshake with a certificate that is not valid for that domain name.

I thought normal behaviour would be that the nginx proxy only listens for requests made to the `mail.domain.tld` hostname, but not other subdomains (including an empty one).

**Comment by Diman0** [comment]
With those settings you should have:
- MX record for domain.tld that points to A record mail.domain.tld.
- A record for mail.domain.tld.

domain.tld should not resolve to mailu (when using those settings).

It looks like nginx is not configured to only listen to the hostnames defined in the HOSTNAMES variable in mailu.env. This only happens for the email proxy.
https://github.com/Mailu/Mailu/blob/master/core/nginx/conf/nginx.conf
I can't see server_name lines for the normal HTTPS traffic.

**Comment by L0g4n** [comment]
Jup, I have those DNS settings (plus the fact that I I have also a wildcard entry for other subdomains that point to the mail server).

Well, I guess I could remove the wildcard DNS entry such that other subdomains never reach the nginx on the mail server. However, I think nginx should be configured to pick up the host name that match the configured mail host name.

**Comment by Diman0** [comment]
Yes it does. I have a reverse proxy in front of mailu where I explicitly have configured server_name directives to make sure only mail.domain.tld (From your example) is redirected to mailu.

I think this could be a handy future for people who do not have a reverse proxy in front of mailu.

**Comment by L0g4n** [comment]
Ah okay, I planned to put my own reverse proxy in the form of nginx in front of mailu, but did not find the time yet 😅
